src/hooks/account/useDeleteAccountMutation.ts

```diff
@@ -1,15 +1,33 @@
+import { HTTPError } from 'ky'
 import { useMutation } from 'react-query'
 import api from 'utils/api'
 
-const deleteAccount = async (accountId: string) => {
-  return await api
+const deleteAccount = (accountId: string) =>
+  api
     .delete(`accounts/${accountId}`)
     //200 HTTP code for delete results in 'unexpected end of JSON input' error, so it must be text
     .text()
-}
 
-const useDeleteAccountMutation = () => {
-  return useMutation(deleteAccount)
+const useDeleteAccountMutation = ({
+  setMessage,
+  successMessage,
+}: {
+  setMessage?: React.Dispatch<React.SetStateAction<string | null>>
+  successMessage?: string
+}) => {
+  return useMutation(deleteAccount, {
+    onError: (err) => {
+      if (err instanceof HTTPError) {
+        const errorResponse = err.response.clone()
+        if (setMessage) errorResponse.json().then((e) => setMessage(e.message))
+      } else if (err instanceof Error) {
+        if (setMessage) setMessage(err.message)
+      }
```

Comment:
If you really wanted to go crazy with the reusability you could create a handleAPIError util function and pass in `err` and `setMessage` so this logic is only defined once.